phylogenetic/rules/construct_phylogeny.smk

```diff
@@ -1,20 +1,54 @@
 """
 This part of the workflow constructs the phylogenetic tree.
 
-REQUIRED INPUTS:
-
-    metadata            = data/metadata.tsv
-    prepared_sequences  = results/prepared_sequences.fasta
-
-OUTPUTS:
-
-    tree            = results/tree.nwk
-    branch_lengths  = results/branch_lengths.json
-
-This part of the workflow usually includes the following steps:
-
-    - augur tree
-    - augur refine
-
 See Augur's usage docs for these commands for more details.
 """
+
+rule tree:
+    """Building tree"""
+    input:
+        alignment = "results/aligned.fasta"
+    output:
+        tree = "results/tree_raw.nwk"
+    log:
+        "logs/tree.txt",
+    benchmark:
+        "benchmarks/tree.txt"
+    shell:
+        """
+        augur tree \
+            --alignment {input.alignment} \
+            --output {output.tree} \
+            2>&1 | tee {log}
+        """
+
+rule refine:
+    """
+    Refining tree
+      - add node information
+    """
+    input:
+        tree = "results/tree_raw.nwk",
+        alignment = "results/aligned.fasta",
+        metadata = "data/metadata.tsv"
+    output:
+        tree = "results/tree.nwk",
+        node_data = "results/branch_lengths.json"
+    log:
+        "logs/refine.txt",
+    benchmark:
+        "benchmarks/refine.txt"
+    params:
+        strain_id = config["strain_id_field"]
+    shell:
+        """
+        augur refine \
+            --tree {input.tree} \
+            --alignment {input.alignment} \
+            --metadata {input.metadata} \
+            --metadata-id-columns {params.strain_id} \
+            --output-tree {output.tree} \
+            --output-node-data {output.node_data} \
+            --root mid_point \
```

Comment:
[Bourhy et al](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3326349/) performed a temporal analysis of N-coding and G-coding regions and found the bat clade to be the outgroup in both cases. We only have 3 bat strains in this analysis whereas they had 25 (N-coding) and 3 (G-coding). The bat samples in our analysis don't form the outrgroup (and I'm not sure they're the same ones they used), so comparing those trees and/or including the bat samples they used and using them as an outgroup would be good ideas to explore.

Reply:
I made the bat clade the outgroup in [5a394cb](https://github.com/nextstrain/rabies/pull/3/commits/5a394cb61d8b507214b095874dba5f6518820fb0). We currently have thousands of bat samples in the tree, [as discussed in slack](https://bedfordlab.slack.com/archives/CFEU0GNNS/p1722901146840109).

Reply:
After [further discussion on slack](https://bedfordlab.slack.com/archives/CFEU0GNNS/p1723073910848769), we decided to stick with midpoint rooting. The Bourhy et al paper does not include the North American raccoon/skunk clade, which changes the shape of the tree. The commit that made the bat clade the outgroup has been dropped.